src/plugins/intel_gpu/src/graph/graph_optimizer/mark_state_init_subgraphs.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "read_value_inst.h"
+#include "pass_manager.h"
+#include <queue>
+
+#include "intel_gpu/graph/program.hpp"
+
+using namespace cldnn;
+
+void mark_state_init_subgraphs::mark_node(program_node* node) {
+    if (node->is_in_state_init_subgraph())
+        return;
+    if (!node->is_type<read_value>())
+        return;
+
+    const auto& variable_id = node->as<read_value>().get_primitive()->variable_id;
+    node->set_state_init_subgraph_id(variable_id);
+
+    // Create a queue for BFS
+    std::queue<program_node*> q;
+    // Enqueue the source node (read_value)
```

Comment:
I think these comments are describing low level implementation. I'd like to suggest to add comments more abstractive and describing the purpose of the code. Low level implementation is expected only if the impl is handling something not general, while this impelmentation is for a general BFS...